packages/maskbook/src/plugins/BarnBridge/SNSAdaptor/index.tsx

```diff
@@ -0,0 +1,41 @@
+import { Plugin, usePostInfoDetails } from '@masknet/plugin-infra'
+import { ChainId } from '@masknet/web3-shared'
+import { Suspense } from 'react'
+import { SnackbarContent } from '@material-ui/core'
+import { URL_PATTERN } from '../constants'
+import { base } from '../base'
+import MaskbookPluginWrapper from '../../MaskbookPluginWrapper'
+import { EthereumChainBoundary } from '../../../web3/UI/EthereumChainBoundary'
+import { BarnBridgeTabContainerView } from '../UI/BarnBridgeTabContainerView'
+
+const isBarnBridgeUrl = (url: string) => URL_PATTERN.test(url)
+
+const sns: Plugin.SNSAdaptor.Definition = {
+    ...base,
+    init(signal) {},
+    PostInspector: function Component() {
+        const link = usePostInfoDetails
+            .postMetadataMentionedLinks()
+            .concat(usePostInfoDetails.postMentionedLinks())
+            .find(isBarnBridgeUrl)
+
+        if (!link) return null
+        return <Renderer url={link} />
+    },
+}
+
+export default sns
+
+function Renderer(props: React.PropsWithChildren<{ url: string }>) {
+    return (
+        <MaskbookPluginWrapper pluginName="BarnBridge">
+            <Suspense fallback={<SnackbarContent message="Mask is loading this plugin..." />}>
+                <EthereumChainBoundary
+                    chainId={ChainId.Mainnet}
```

Comment:
It should read the current chain id instead of dead-code Mainnet. If given an invalid chain id, the boundary will reveal the switch chain panel for you.

Reply:
Ah will do good callout. This plugin doesn't really use any contract info -- although i guess it does use wallet address. I'll do it the right way

Reply:
So taking a look at it -- `PoolTogether` also hardcodes the chain id. Since I use no web3 calls in the entire plugin i don't see a reason why it makes sense to do this. For a future update I will need to (for the YieldFarming tab) when that features come i can integrate web3 and the token contracts for the correct chain support.